daemon/cmd/daemon_main.go

```diff
@@ -1455,6 +1456,18 @@ func (d *Daemon) instantiateAPI() *restapi.CiliumAPI {
 	return restAPI
 }
 
+func initSockmapOption() {
+	if option.Config.SockopsEnable {
+		if h := probes.NewProbeManager().GetHelpers("sk_msg"); h != nil {
+			if _, ok := h["bpf_msg_redirect_hash"]; ok {
+				return nil
```

Comment:
The `return nil` here might be the cause for the build error. Simply `return` should work?

Note: Instead of having all the code of that function if the `if option.Config.SockopsEnable` block, you could also return early if the option is not set and avoid indenting all the rest:

``` go
if !option.Config.SockopsEnable {
        return
}
if h := probes.NewProbeManager()...
```
It usually makes the code a bit easier to read, I find.

Reply:
@qmonnet The probe still needs to be fixed anyway ;-)

Reply:
ok will fix it. @pchaigno is the probe fix that you mentioned above related to this patch or we need to handle it in a different PR>

Reply:
Related to this patch. Please see https://github.com/cilium/cilium/pull/10941#discussion_r407888034 and https://github.com/cilium/cilium/pull/10941#discussion_r408650529.